test/tests/functional/pbs_ralter.py

```diff
@@ -0,0 +1,1112 @@
+# coding: utf-8
+
+# Copyright (C) 1994-2017 Altair Engineering, Inc.
+# For more information, contact Altair at www.altair.com.
+#
+# This file is part of the PBS Professional ("PBS Pro") software.
+#
+# Open Source License Information:
+#
+# PBS Pro is free software. You can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU Affero General Public License for more
+# details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# The PBS Pro software is licensed under the terms of the GNU Affero General
+# Public License agreement ("AGPL"), except where a separate commercial license
+# agreement for PBS Pro version 14 or later has been executed in writing with
+# Altair.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software - under a
+# commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+from tests.functional import *
+
+
+class TestPbsResvAlter(TestFunctional):
+
+    """
+    This test suite is for testing the reservation times modification feature.
+    """
+
+    # Class variables
+    tzone = None
+    # These must be equal to the declarations in pbs_error.h
+    PBSE_UNKRESVID = 236
+    PBSE_RESV_NOT_EMPTY = 74
+    PBSE_STDG_RESV_OCCR_CONFLICT = 75
+    PBSE_INCORRECT_USAGE = 2
+    fmt = "%a %b %d %H:%M:%S %Y"
+
+    def setUp(self):
+
+        TestFunctional.setUp(self)
+        self.server.set_op_mode(PTL_CLI)
+        # Set PBS_TZID, needed for standing reservation.
+        if 'PBS_TZID' in self.conf:
+            self.tzone = self.conf['PBS_TZID']
+        elif 'PBS_TZID' in os.environ:
+            self.tzone = os.environ['PBS_TZID']
+        else:
+            self.logger.info('Timezone not set, using Asia/Kolkata')
+            self.tzone = 'Asia/Kolkata'
+
+        a = {'resources_available.ncpus': 4}
+        self.server.create_vnodes('vnode', a, num=2, mom=self.mom,
+                                  usenatvnode=True)
+
+        rc = self.server.manager(MGR_CMD_SET, SERVER, {
+            'log_events': 4095}, expect=True)
+        self.assertTrue(rc)
+
```

Comment:
self.assertTrue(rc) is not required if you are doing "expect=True". Any exception will be caught in case qmgr set command fails.

Reply:
done!